store/fifo_store.go

```diff
@@ -0,0 +1,96 @@
+package store
+
+import (
+	"container/list"
+	"sync"
+
+	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/prometheus/internalmetrics"
+)
+
+// FIFOStore uses a FIFO approach storage and is safe for concurrent use.
+type FIFOStore[K comparable, V any] struct {
+	capacity int
+	lock     sync.RWMutex
+	name     string // Used for metrics
+	order    *list.List
+	Store[K, V]
+}
+
+func NewFIFOStore[K comparable, V any](store Store[K, V], capacity int, name string) *FIFOStore[K, V] {
+
+	f := &FIFOStore[K, V]{
+		capacity: capacity,
+		Store:    store,
+		order:    list.New(),
+		name:     name,
+	}
+	return f
+}
+
+// Get returns the value associated with the given key or an error.
+func (f *FIFOStore[K, V]) Get(key K) (V, bool) {
+	internalmetrics.GetCacheRequestsTotalMetric(f.name).Inc()
+	elem, exists := f.Store.Get(key)
+	if !exists {
+		var value V
+		log.Tracef("[FIFO store] Element doesnt exist: %v", key)
+		return value, false
+	}
+
+	log.Tracef("[FIFO store] Returning from cache: %v", key)
+	internalmetrics.GetCacheHitsTotalMetric(f.name).Inc()
+	return elem, true
+}
+
+// Set
+func (f *FIFOStore[K, V]) Set(key K, value V) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	_, exists := f.Store.Get(key)
+
+	// Remove older
+	if f.order != nil && f.order.Len() >= f.capacity && !exists {
+		oldest := f.order.Front()
+		if oldest != nil {
+			f.order.Remove(oldest)
+			f.Store.Remove(oldest.Value.(K))
+		}
+	}
+
+	if !exists {
+		f.order.PushBack(key)
+	}
+	f.Store.Set(key, value)
+}
+
+// Set
+func (f *FIFOStore[K, V]) Remove(key K) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	for e := f.order.Front(); e != nil; e = e.Next() {
+		if e.Value == key {
+			f.order.Remove(e)
+			break
+		}
+	}
```

Comment:
Internally `f.order.Remove` should handle iterating to find the element so maybe this can just be
```suggestion
	f.order.Remove(list.Element{Value: key})
```